core/worker.go

```diff
@@ -1713,87 +1686,113 @@ func (w *worker) prepareWork(genParams *generateParams, wo *types.WorkObject) (*
 			newWo.Header().SetExchangeRate(params.ExchangeRate)
 		} else {
 
-			// use the parent exchange rate
-			parentExchangeRate := parent.ExchangeRate()
-
-			// There are 6 steps to the exchange rate calculation
-			// 1. Convert the amounts using the parent exchange rate or the exchange
-			// rate set in the statedb
-			// 2. Using the parent K Quai apply a discount to the converted amount
-			// 3. Calculate the new exchange rate for this block
-			// 4. Convert the amounts using this newly computed exchange rate
-			// 5. Apply the K Quai discount using the new exchange rate
-			// 6. Apply the quadratic Conversion Flow discount
-
 			///////// Step 1 //////////
-			var conversionFlowAmount = parent.ConversionFlowAmount()
-
 			// Read inbound etxs, this needs to be calculated for the pending manifests
 			inboundEtxs := rawdb.ReadInboundEtxs(w.workerDb, parent.Hash())
+			if inboundEtxs == nil {
+				return nil, errors.New("cannot find the inbound etxs for parent")
```

Comment:
Are you absolutely sure it's impossible to have zero inbound etxs?

Reply:
its not zero, its only if we havent written, in the ReadInboundEtxs method we return empty list in case its zero